dbt/models/proximity/proximity.dist_pin_to_traffic_master.sql

```diff
@@ -0,0 +1,340 @@
+        ARBITRARY(xy.daily_traffic) AS nearest_other_road_daily_traffic,
+        ARBITRARY(xy.speed_limit) AS nearest_other_road_speed_limit,
+        ARBITRARY(xy.surface_type) AS nearest_other_road_surface_type,
+        ARBITRARY(xy.lanes) AS nearest_other_road_lanes
+    FROM distinct_pins AS pcl
+    INNER JOIN ( {{ dist_to_nearest_geometry('traffic_other') }} ) AS xy
+        ON pcl.x_3435 = xy.x_3435
+        AND pcl.y_3435 = xy.y_3435
+    GROUP BY pcl.pin10, xy.year
+),
+
+-- Join all nearest roads by pin10 and year
+final_aggregation AS (
+    SELECT
+        COALESCE(
+            minor.pin10, interstate.pin10, freeway.pin10,
+            local_road.pin10, collector.pin10
+        ) AS pin10,
+        minor.nearest_minor_road_name,
+        minor.nearest_minor_road_dist_ft,
+        minor.nearest_minor_road_data_year,
+        minor.nearest_minor_road_daily_traffic,
+        minor.nearest_minor_road_speed_limit,
+        minor.nearest_minor_road_surface_type,
+        minor.nearest_minor_road_lanes,
+        interstate.nearest_interstate_road_name,
+        interstate.nearest_interstate_road_dist_ft,
+        interstate.nearest_interstate_road_data_year,
+        interstate.nearest_interstate_road_daily_traffic,
+        interstate.nearest_interstate_road_speed_limit,
+        interstate.nearest_interstate_road_surface_type,
+        interstate.nearest_interstate_road_lanes,
+        freeway.nearest_freeway_road_name,
+        freeway.nearest_freeway_road_dist_ft,
+        freeway.nearest_freeway_road_data_year,
+        freeway.nearest_freeway_road_daily_traffic,
+        freeway.nearest_freeway_road_speed_limit,
+        freeway.nearest_freeway_road_surface_type,
+        freeway.nearest_freeway_road_lanes,
+        local_road.nearest_local_road_name,
+        local_road.nearest_local_road_dist_ft,
+        local_road.nearest_local_road_data_year,
+        local_road.nearest_local_road_daily_traffic,
+        local_road.nearest_local_road_speed_limit,
+        local_road.nearest_local_road_surface_type,
+        local_road.nearest_local_road_lanes,
+        collector.nearest_collector_road_name,
+        collector.nearest_collector_road_dist_ft,
+        collector.nearest_collector_road_data_year,
+        collector.nearest_collector_road_daily_traffic,
+        collector.nearest_collector_road_speed_limit,
+        collector.nearest_collector_road_surface_type,
+        collector.nearest_collector_road_lanes,
+        other.nearest_other_road_name,
+        other.nearest_other_road_dist_ft,
+        other.nearest_other_road_data_year,
+        other.nearest_other_road_daily_traffic,
+        other.nearest_other_road_speed_limit,
+        other.nearest_other_road_surface_type,
+        other.nearest_other_road_lanes,
+        COALESCE(
+            minor.year, interstate.year, freeway.year,
+            local_road.year, collector.year, other.year)
+            AS year
+    FROM nearest_minor AS minor
```

Comment:
I think rather than doing all of this coalescing for `pin10` you could just start with `spatial.parcel` as your base for pins and years and left join everything on to that. My guess is that this will be less resource intensive.